### PR TITLE
Fix tagging related things

### PR DESCRIPTION
**Pull request by haikuoliu**
### Summary

1. Don't pass tags with RCI call when empty
2. Remove tagging in README and CHANGALOG
3. Add missing change log for pid/ipc
4. Add a test for the case where tags are empty
5. Skip tagging func test

### Testing

- [ ] Builds on Linux (`make release`)
- [ ] Builds on Windows (`go build -out amazon-ecs-agent.exe ./agent`)
- [ ] Unit tests on Linux (`make test`) pass
- [ ] Unit tests on Windows (`go test -timeout=25s ./agent/...`) pass
- [ ] Integration tests on Linux (`make run-integ-tests`) pass
- [ ] Integration tests on Windows (`.\scripts\run-integ-tests.ps1`) pass
- [ ] Functional tests on Linux (`make run-functional-tests`) pass
- [ ] Functional tests on Windows (`.\scripts\run-functional-tests.ps1`) pass

### Licensing

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: agent/api/ecsclient/client.go [hunk 1]
```diff
@@ -157,7 +157,9 @@ func (client *APIECSClient) registerContainerInstance(clusterRef string, contain
 	// Add additional attributes such as the os type
 	registrationAttributes = append(registrationAttributes, client.getAdditionalAttributes()...)
 	registerRequest.Attributes = registrationAttributes
-	registerRequest.Tags = tags
+	if tagsEnabled && len(tags) > 0 {
```
Comment (adnxn) [inline agent/api/ecsclient/client.go]: i don't think we need the `tagsEnabled` parameter. we can add an else block here for setting this field, lets try to keep the interfaces smaller if we can help it.

Comment (haikuoliu) [reply agent/api/ecsclient/client.go]: Just removed the parameter

**Comment by haikuoliu** [comment]
The failed func tests are v3 endpoint related tests, which are already fixed in #1660, and `TestAgentIntrospectionValidator` which is another known flaky test. Going to merge this PR.
